internal/api/option/error.go

```diff
@@ -0,0 +1,91 @@
+package option
+
+import (
+	"context"
+	"net/http"
+
+	"connectrpc.com/connect"
+	"github.com/pkg/errors"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var (
+	_ connect.Interceptor = &errorInterceptor{}
+)
+
+type errorInterceptor struct{}
+
+func newErrorInterceptor() connect.Interceptor {
+	return &errorInterceptor{}
+}
+
+func (i *errorInterceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
+	return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		res, err := next(ctx, req)
+		if err != nil {
+			return nil, i.wrapError(err)
+		}
+		return res, nil
+	}
+}
+
+func (i *errorInterceptor) WrapStreamingClient(
+	next connect.StreamingClientFunc,
+) connect.StreamingClientFunc {
+	// TODO: Support streaming client when necessary
+	return next
+}
+
+func (i *errorInterceptor) WrapStreamingHandler(
+	next connect.StreamingHandlerFunc,
+) connect.StreamingHandlerFunc {
+	return func(ctx context.Context, conn connect.StreamingHandlerConn) error {
+		if err := next(ctx, conn); err != nil {
+			return i.wrapError(err)
+		}
+		return nil
+	}
+}
+
+func (i *errorInterceptor) wrapError(err error) error {
```

Comment:
Nit on naming. "wrap" makes sense for things like `WrapStreamingClient` and other functions above, because those functions are, indeed, wrapping a client, handler, etc. with additional functionality.

I'm not sure if it makes sense here with `wrapError`, because that's not what this function does -- or at least it doesn't in all cases. I think it's function would be more clear if it were named something like `toConnectError`.

Reply:
Tiniest nit -- the receiver `i` isn't used in this function. Eliminating it would make it the tiniest bit easier to test it. To be clear, I wouldn't care if this _didn't_ change, but I'm only pointing it out because I think it's a good practice to structure code such that it's maximally easy to test.